src/managers/kyuubi.py

```diff
@@ -48,14 +48,24 @@ def update(
         set_zookeeper_none: bool = False,
         set_tls_none: bool = False,
         force_restart: bool = False,
-    ):
+    ) -> None:
         """Update Kyuubi service and restart it."""
         metastore_db_info = None if set_metastore_db_none else self.context.metastore_db
         auth_db_info = None if set_auth_db_none else self.context.auth_db
         service_account_info = None if set_service_account_none else self.context.service_account
         zookeeper_info = None if set_zookeeper_none else self.context.zookeeper
         tls_info = None if set_tls_none else self.context.tls
 
+        # auth is mandatory
```

Comment:
I think having this check here is a bit of an overkill. For example, we don't do this if Integration Hub hasn't been related, or for example, the object storage is not available. And, we will put the charm in blocked state, anyway.

This check potentially will result in us adding further checks in the future for all other "blocking" checks (eg, integration hub, object storage, etc.) and thus potentially we will have to repeat the validation logic for all blocking conditions at two places in the charm.

Reply:
We need to distinguish between "blocked state" and "workflow is not running" IMO. If the charm is in blocked status because of a missing object storage and you try to use it only to get an error, that's on you.
If the charm is in blocked status because of missing auth but you can still interact with it using beeline that's a whole different problem.

Were you thinking of something else? I might have misunderstood

Reply:
Thanks for the clarification, Alex. I agree with it.